### PR TITLE
fixes GH-144: add support for `require-sri-for` directive

### PR DESCRIPTION
**Pull request by shekyan**
- supports `require-sri-for`
- tests
- refactored sandbox and referrer directive value handling to use generic
RFC7230Token class.

### CODE REVIEW

File: src/test/java/com/shapesecurity/salvation/LocationTest.java [hunk 1]
```diff
@@ -306,6 +306,54 @@
         assertEquals(27, notices.get(1).endLocation.offset);
     }
 
+    @Test public void testWarningLocationRequireSriFor() {
+        ArrayList<Notice> notices = new ArrayList<>();
+
+        ParserWithLocation.parse("require-sri-for", "https://origin", notices);
```
Comment (michaelficarra) [inline src/test/java/com/shapesecurity/salvation/LocationTest.java]: This should be invalid.

File: src/main/java/com/shapesecurity/salvation/Parser.java [hunk 2]
```diff
@@ -34,6 +34,10 @@
         new DirectiveParseException("Invalid referrer-token");
     private static final DirectiveParseException INVALID_REPORT_TO_TOKEN =
         new DirectiveParseException("Invalid report-to token");
+    private static final DirectiveParseException INVALID_REQUIRE_SRI_FOR_TOKEN_LIST =
+            new DirectiveParseException("Invalid require-sri-for-token list");
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/Parser.java]: Remove these dashes before `token`.

File: src/test/java/com/shapesecurity/salvation/ParserTest.java [hunk 3]
```diff
@@ -744,6 +744,50 @@
...
+        assertEquals("require-sri-for script style", p.show());
+
+        notices.clear();
+        p = parseWithNotices("require-sri-for Script    StYlE", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals(0, notices.size());
+        assertEquals("require-sri-for script style", p.show());
+
+        notices.clear();
+        p = parseWithNotices("require-sri-for script script", notices);
```
Comment (michaelficarra) [inline src/test/java/com/shapesecurity/salvation/ParserTest.java]: Duplicates should create a warning.

File: src/test/java/com/shapesecurity/salvation/LocationTest.java (hunk 1 above)
Comment (shekyan) [inline src/test/java/com/shapesecurity/salvation/LocationTest.java]: If we don't change the spec grammar, this remains a valid policy, so I think warning is appropriate here.

**Comment by shekyan** [comment]
Addressed comments.

**Pull request merged**